app/forms/upload_form.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class UploadForm
+  include ActiveModel::Validations
+
+  attr_reader :search_engine, :csv_file
+
+  validates :search_engine, presence: true
+  validates :search_engine, inclusion: { in: SearchResult.search_engines.keys }
+  validates :csv_file, presence: true
+
+  validate :valid_content_type, if: -> { csv_file.present? }
+  validate :valid_count, if: -> { csv_file.present? }
+
+  def initialize(params)
+    @search_engine = params[:search_engine]
+    @csv_file = params[:csv_file]
+  end
+
+  def valid_content_type
+    return if csv_file.content_type == 'text/csv'
+
+    add_error :csv_file_invalid_extension
+  end
+
+  def valid_count
+    return if CSV.read(csv_file.tempfile).count.between?(1, 1000)
```

Comment:
The Mac count should be 100 and the value should be defined as a constant

Reply:
I'll move it as a constant, however the requirements specify 1000 (not 100)

>  The uploaded file can be of any size from 1 to 1,000 keywords.